src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -27,25 +31,65 @@ namespace CoreNodeModels.Input
     [OutPortDescriptions("Selected Color.")]
     public class ColorPalette : NodeModel
     {
-        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
-        public DSColor dsColor
+        private DSColor dscolor = DSColor.ByARGB(255, 0, 0, 0);
+        private DSColor prevColor = DSColor.ByARGB(255, 0, 0, 0);
+        private bool Isundo = false;
```

Comment:
`IsUndo` gets used when an Undo or Redo command is actually doing the property setting - when an Undo or Redo is effecting the property state we want to ignore this state, and not propagate an update, as this will get recorded as an undo state... which is not how the undo recorder works.

What about calling it - `DoNotRecordForUndo`